src/Init/Data/String/Basic.lean

```diff
@@ -89,13 +123,38 @@ def utf8SetAux (c' : Char) : List Char → Pos → Pos → List Char
   | c::cs, i, p =>
     if i = p then (c'::cs) else c::(utf8SetAux c' cs (i + c) p)
 
+/--
+Replaces the character at a specified position in a string with a new character.
+
+If the position is invalid, returns the string unchanged.
+* `"abc".set ⟨1⟩ 'B' = "aBc"`
+* `"abc".set ⟨3⟩ 'D' = "abc"`
+* `"L∃∀N".set ⟨1⟩ 'X' = "L∃∀N" -- invalid position as '∃' is a mutli-byte character`
+
+If both the replacement character and the replaced character are ASCII characters and the string is not shared, destructive updates are used.
+-/
 @[extern "lean_string_utf8_set"]
 def set : String → (@& Pos) → Char → String
   | ⟨s⟩, i, c => ⟨utf8SetAux c s 0 i⟩
 
+/--
+Applies `f` to the character at position `p` in the string `s`.
+
+If `p` is an invalid position, returns the string unchanged.
+* `abc.modify ⟨1⟩ Char.toUpper = "aBc"`
+* `abc.modify ⟨3⟩ Char.toUpper = "abc"`
+-/
 def modify (s : String) (i : Pos) (f : Char → Char) : String :=
   s.set i <| f <| s.get i
 
+/--
+Returns the next position in a string after position `p`.
+
+If `p` is not a valid position or `p = s.endPos`, the result is unspecified.
+* `"abc".next ⟨1⟩ = String.Pos.mk 2`
+* `"abc".next ⟨3⟩ = String.Pos.mk 4`
+* `"L∃∀N".next ⟨1⟩ => String.Pos.mk 4 -- '∃' is a mutli-byte character`
```

Comment:
These examples could use a bit more context. The first one is showing the expected functioning of the function, and the other two are the two exceptional cases, but this is not very clear from the description.

Reply:
I split the examples for String.next into two sections: "Examples" and "Cases where the result is unspecified." I have also included explanation of the confusing cases.